### DISCUSSION TITLE
listen to changes without controlling the internal table state

### DISCUSSION THREAD

**Discussion by astahmer** [TanStack/table#4238] [Q&A]
Hey, I might have missed in the docs/examples but I can't find a way to listen without controlling the table's internal state. The methods `onSortingChange` | `onExpandedChange` | `onPaginationChange` all implies having to manage the state/update, what if I just wanted to be notified when that happens, is it possible right now ?

**Comment by KevinVandy** [comment]
It is best to control the state and hoist it up to where you want to have access to it. If you don't want to create a useState for every individual state, you can create 1 state, setState instead. See the [Fully Controlled Example](https://tanstack.com/table/v8/docs/examples/react/fully-controlled)

You could also technically make a useEffect that listens to `table.getState().sorting`, or whatever state and then perform side effects there.

**Reply by astahmer** [reply]
Yeah I was considering going the `useEffect` way, thanks for confirming it as a viable solution !
